src/NUnitFramework/framework/Assert.cs

```diff
@@ -96,13 +102,20 @@ public static new void ReferenceEquals(object a, object b)
         /// </summary>
         /// <param name="message">The message to initialize the <see cref="AssertionException"/> with.</param>
         /// <param name="args">Arguments to be used in formatting the message</param>
-        static public void Pass(string message, params object[] args)
+        public static void Pass(string message, params object[] args)
         {
-            if (message == null) message = string.Empty;
-            else if (args != null && args.Length > 0)
-                message = string.Format(message, args);
+            Assert.Pass(BuildExceptionMessageFunc(message, args));
+        }
 
-            throw new SuccessException(message);
+        /// <summary>
+        /// Throws a <see cref="SuccessException"/> with the message and arguments 
+        /// that are passed in. This allows a test to be cut short, with a result
+        /// of success returned to NUnit.
+        /// </summary>
+        /// <param name="getExceptionMessage">A function to build the message included with the Exception</param>
+        public static void Pass(Func<string> getExceptionMessage)
```

Comment:
I felt I should add these for consistency.  I was adding an overload to `Assert.That<TActual>` that accepts a `Func<string>` so you could dynamically build an Exception Message.  I am not certain of all of the use cases, so that's why I wanted to leave it open to usage - which is why I added it to the Pass/Fail/Inconclusive/Ignore.

That being said, I'm not emotionally attached to it.  :-)

If it is removed, we can move that helper method to the Assert.That class (if it is still used), and revert the file changes.